HIP/hip-657.md

```diff
@@ -0,0 +1,115 @@
+---
+hip: 657
+title: Mutable metadata fields for dynamic NFTs
+author: Patches Punks <@HGraphPunks>
+working-group: Jason Fabritz <@bugbytesinc>, Sean Cuscus <@seancuscus>, Cooper Kunz <@Cooper-Kunz>, May Chan <@rocketmay>, Burstall Stowerling <@Burstall>, Ian Holsman <@web3-nomad>, Ashe Oro <@Ashe-Oro>, Michael Garber <@mgarbs>, Paul Sullivan <@paulgs9988>
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Review
+created: 2023-01-04
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/607
+updated: 2023-01-21, 2023-01-24
+---
+
+
+## Abstract
+
+
+Currently, there is no native way to update the metadata of a non-fungible token (NFT) on the Hedera network. This HIP proposes the creation of a new key called the METADATA key, which will allow updating NFT metadata. This will provide the Hedera network with functional parity with other networks and enable developers to create dynamic NFTs with new interactions.
+
+
+## Motivation
+
+
+Dynamic NFTs are a valuable addition to the world of non-fungible tokens because they allow for more interactive and immersive experiences. With the ability to update the metadata of an NFT, developers can create game pieces that gain new equipment and stats, legal agreements that reflect changing parameters, or even NFTs that update their images based on the time of day. Dynamic NFTs also open up the possibility for NFTs to be used in a wider range of industries and applications, such as supply chain tracking, real estate, among others. The potential for dynamic NFTs is numerous, and their introduction will greatly enhance the capabilities of the NFT ecosystem.
```

Comment:
the network won't track the various states of the metadata (ie the history of the metadata values) but rather than the metadata was updated since it will be an on-chain transaction.

Reply:
Wouldnt it be good to know what the updates were though? Let's say we represent a property as an nft and all its attributes are defined in the metadata (price, floorplan, additions, etc) When updates happen to the property, it would be good to look onchain to verify to see what they were, no?

Reply:
if it's possible to query previous values of `metadata` then that would be ideal, but i'm not certain how that would work and don't have a solution to propose.

Reply:
Yea, true. This part would be out of scope for this hip. We will discuss a separate mirror node hip at a later date

Reply:
I don't think this is outside the scope of this HIP. We should take into consideration Mirror Node exposing this data. @steven-sheehy looping you in. Do you have any thoughts or see any concerns with having to surface multiple versions of NFT metadata?

Reply:
ping @steven-sheehy . I believe this is the last remaining open discussion for this HIP. 

Can/should the mirror node track the change history for metadata field values? IMO it would be extremally helpful if possible. 

Here's an example: 

My house is represented as an NFT. I replace the roof with a new 30-year warranty which causes and update to my NFT's metadata. Being able to review the history of my home's various states throughout time helps new buyers audit work done on my house.